modules/auxiliary/scanner/http/owa_ews_login.rb

```diff
@@ -0,0 +1,169 @@
+##
+# This module requires Metasploit: http://metasploit.com/download
+# Current source: https://github.com/rapid7/metasploit-framework
+##
+
+require 'msf/core'
+require 'rex/proto/ntlm/message'
+require 'rex/proto/http'
+require 'metasploit/framework/credential_collection'
+
+class Metasploit3 < Msf::Auxiliary
+
+  include Msf::Auxiliary::Report
+  include Msf::Auxiliary::AuthBrute
+  include Msf::Auxiliary::Scanner
+
+  def initialize
+    super(
+      'Name'           => 'OWA Exchange Web Services (EWS) Login Scanner',
+      'Description'    => %q{
+        This module attempts to log in to the Exchange Web Services, often
+        exposed at https://example.com/ews/, using NTLM authentication. This
+        method is faster and simpler than traditional form-based logins.
+
+        In most cases, all you need to set is RHOSTS and some combination of
+        user/pass files; the autodiscovery should find the location of the NTLM
+        authentication point as well as the AD domain, and use them accordingly.
+      },
+      'Author'         => 'Rich Whitcroft',
+      'License'        => MSF_LICENSE,
+      'DefaultOptions' => { 'SSL' => true, 'VERBOSE' => false }
+    )
+
+    register_options(
+      [
+        OptBool.new('AUTODISCOVER', [ false, "Automatically discover domain URI", true ]),
+        OptString.new('AD_DOMAIN', [ false, "The Active Directory domain name", nil ]),
+        OptString.new('TARGETURI', [ false, "The location of the NTLM service", nil ]),
+        OptInt.new('RPORT', [ true, "The target port", 443 ])
+      ], self.class)
+  end
+
+  def run_host(ip)
+    cli = Rex::Proto::Http::Client.new(datastore['RHOSTS'], datastore['RPORT'], {}, datastore['SSL'], datastore['SSLVersion'], nil, '', '')
```

Comment:
I tried this originally but couldn't get it to work. I need to check the response code of the request which gets abstracted away by HttpClient I think, since `code != 401` isn't a typical check for success, but is required here.